bundle/jsky.app.ot/src/main/java/jsky/app/ot/viewer/SPElevationPlotPlugin.java

```diff
@@ -274,14 +281,26 @@ private LegendItemCollection getLegendItems() {
         final LegendItemCollection lic;
         switch (_colorCodeTrajectories) {
             case _COLOR_CODE_OBS:
-                lic = createLegendItems(obs -> obs.getObservationIDAsString("unknown"));
+                lic = createLegendItems(
+                        Comparator.comparing(ISPObservation::getObservationID),
+                        obs -> obs.getObservationIDAsString("unknown")
+                );
                 break;
 
             case _COLOR_CODE_PROG:
-                lic = createLegendItemsOpt(obs -> ImOption.apply(obs.getProgramID()).map(SPProgramID::stringValue));
+                lic = createLegendItemsOpt(
+                        (o1,o2) -> {
+                            final SPProgramID id1 = o1.getProgramID();
+                            final SPProgramID id2 = o2.getProgramID();
+                            if (id1 == null) return (id2 == null) ? 0 : 1;
+                            return (id2 == null) ? -1 : id1.compareTo(id2);
```

Comment:
There seems to be a `Comparator.nullsFirst` that handles `null`.

Reply:
Nice! Did not know about that. I'll make the change.